index.js

```diff
@@ -160,17 +160,14 @@
             // wrap it to a react class automatically
             if (typeof componentClass === "function" && !componentClass.prototype.render && !componentClass.isReactClass && !React.Component.isPrototypeOf(componentClass)) {
                 return observer(React.createClass({
-                    displayName: componentClass.name,
-                    propTypes: componentClass.propTypes,
-                    getDefaultProps: function() {
-                      return componentClass.defaultProps;
-                    },
-                    render: function() {
-                        return componentClass.call(this, this.props);
-                    }
+                    displayName:     componentClass.displayName || componentClass.name,
+                    propTypes:       componentClass.propTypes,
+                    contextTypes:    componentClass.contextTypes,
+                    getDefaultProps: function() { return componentClass.defaultProps; },
+                    render:          function() { return componentClass.call(this, this.props, this.context); }
```

Comment:
here are the actual changes:

1. added `contextTypes`
2. added `this.context` as 3rd parameter in the `render()` function
3. added `componentClass.displayName` as another alternative to explicitly name stateless `render` functions